File: Cpp/BFS.c
#include <stdbool.h>
#include <stdio.h>
#include "Graph.h"

void BFS(Graph* g, int s){

	// Visited list
	bool visited[MAX_VERTICES];
	for(int i=0; i<g->V; i++){
		visited[i] = false;
	}

	// Queue for BFS
	int queue[MAX_VERTICES];
	int front = 0, rear = 0;

	// Mark current node as visited
	visited[s] = true;
	queue[rear++] = s;

	while(front != rear){
		s = queue[front++];
		printf("%d ", s);

		for(int adjs=0; adjs<g->V; adjs++){
			if(g->adj[s][adjs] && !visited[adjs]){
				visited[adjs] = true;
				queue[rear++] = adjs;
			}
		}
	}
}

File: Python/BFS.py
from collections import deque

def bfs(graph, start):
    visited = set()
    queue = deque([start])
    visited.add(start)

    while queue:
        current = queue.popleft()
        print(current, end=" ")

        for neighbor in graph[current]:
            if neighbor not in visited:
                queue.append(neighbor)
                visited.add(neighbor)

graph = {
    'A': ['B', 'C'],
    'B': ['A', 'D', 'E'],
    'C': ['A', 'F', 'G'],
    'D': ['B'],
    'E': ['B', 'H'],
    'F': ['C'],
    'G': ['C'],
    'H': ['E']
}

start = 'A'
print("BFS starting from vertex", start_vertex)
bfs(graph, start_vertex)